pkg/resolution/resolver/git/config.go

```diff
@@ -43,3 +51,54 @@ const (
 	// APISecretNamespaceKey is the config map key for the token secret's namespace
 	APISecretNamespaceKey = "api-token-secret-namespace"
 )
+
+type GitResolverConfig struct {
+	ScmTokens map[string]ScmInfo
+}
+
+type ScmInfo struct {
+	Timeout            string `json:"fetch-timeout"`
+	URL                string `json:"default-url"`
+	Revision           string `json:"default-revision"`
+	Org                string `json:"default-org"`
+	ServerURL          string `json:"server-url"`
+	SCMType            string `json:"scm-type"`
+	APISecretName      string `json:"api-token-secret-name"`
+	APISecretKey       string `json:"api-token-secret-key"`
+	APISecretNamespace string `json:"api-token-secret-namespace"`
+}
+
+func GetGitConfig(ctx context.Context) GitResolverConfig {
+	var scmInfo interface{} = &ScmInfo{}
+	structType := reflect.TypeOf(scmInfo).Elem()
+	gitResolverConfig := GitResolverConfig{ScmTokens: map[string]ScmInfo{}}
+	conf := framework.GetResolverConfigFromContext(ctx)
+	for k, v := range conf {
+		var tokenIdentifier, tokenKey string
+		key := strings.Split(k, ".")
+		if len(key) >= 3 && key[0] == "identifier" {
+			tokenKey = strings.Join(key[2:], ".")
```

Comment:
this is to handle the case `identifier.test.default-url` like these key in configmap

Reply:
What I meant here is that you only need re-join `key[2:]` if there are more dots than expected, e.g. if we had something like `identified.test.default.url`, which we don't and don't want to have.
If thought it would be better to fail here and say that the config is invalid.

If you let it through, as you said it won't match any of the expected config, and it will be silently ignored, which may lead users to think they have configured something while they actually didn't.

I think it's fine to keep this as it is here, as long as we fail later on for unrecognised settings.

Reply:
did the handling for invalid key format